loopy/tools.py

```diff
@@ -604,4 +612,128 @@ def intern_frozenset_of_ids(fs):
     return frozenset(intern(s) for s in fs)
 
 
-# vim: foldmethod=marker
+def _verify_generated_reproducer_is_equivalent(python_code, kernel, var_name):
+    reproducer_variables = {}
+    exec(python_code, reproducer_variables)
+    assert reproducer_variables[var_name] == kernel
```

Comment:
Move the `assert` out (so that in the `python -O` case, we don't pay the cost of the `exec`?)

Reply:
Done, thanks!